src/key_store/mod.rs

```diff
@@ -1,3 +1,39 @@
+//! A storage solution for private key material.
+//!
+//! This module provides access to the `KeyStore` struct, which manages the
+//! storage of `CredentialBundle` and `KeyPackageBundle` instances. The
+//! development of this module is tracked in #337, which also includes a
+//! roadmap.
+//!
+//! The current key store enables the storage of `CredentialBundle` instances,
+//! and grants access to `CredentialBundle` references via the
+//! `SignaturePublicKey` of the corresponding `Credential`.
+//!
+//! A `KeyStore` is meant to be used across multiple `ManagedGroup` instances to
+//! allow sharing the same `CredentialBundle`. If this is not desired, multiple
+//! `KeyStore` instances can be used across groups.
+//!
+//! # Example
+//!
+//! A simple example for the generation and the retrieval of a
+//! `CredentialBundle`.
+//!
+//! ``` let mut key_store = KeyStore::default();
+//!
+//! // Generate credential bundles
+//! let alice_credential = key_store
+//!     .generate_credential(
+//!         "Alice".into(),
+//!         CredentialType::Basic,
+//!         SignatureScheme::ED25519,
+//!     )
+//!     .unwrap()
+//!     .clone();
```

Comment:
Why is the credential cloned after creation?

Reply:
If we don't clone it, we have a dangling mutable reference to the KeyStore and can't borrow immutably for the subsequent call to `get_credential_bundle`. I plan to fix this with the next PR.

Reply:
Ok but this makes the entire key store pretty much unusable if every reference it returns has to be cloned.

Reply:
I'm a bit confused. I thought I made clear that this was a limitation of the current key store and that a solution (in the shape of RefCells) comes with the next PR. Note, that the cloning is only necessary for `Credential`s. For `CredentialBundle`s it's fine, because getting them only requires an immutable reference.

I feel like we always get back to this point. But since the `RefCell` solution is in all likelyhood going to be ready until next week, I'll probably end up closing this PR for a version that doesn't require cloning anyway.